### PR TITLE
Improvement femr10 display user names

### PR DESCRIPTION
**Pull request by akrebs1202**
In a patient's history, the nurse, physician, and pharmacist will be displayed with the users full name instead of email address.

### CODE REVIEW

File: app/femr/business/services/system/EncounterService.java [hunk 1]
```diff
@@ -89,7 +89,10 @@ public EncounterService(IMissionTripService missionTripService,
                     .where()
                     .eq("email", patientEncounterItem.getNurseEmailAddress());
 
-            IUser nurseUser = userRepository.findOne(nurseQuery);
+            ExpressionList<User> nurseQuery1 = QueryProvider.getUserQuery()
+                    .where()
+                    .eq("email", patientEncounterItem.getNurseFullName()); //Andrew Change
+            IUser nurseUser = userRepository.findOne(nurseQuery1);
```
Comment (kdunlap) [inline app/femr/business/services/system/EncounterService.java]: I don't think we want this change. When testing I got a null pointer exception. I think we need to delete nurseQuery1 and go back to nurseQuery

We only know the nurse user's email address at this point, so we need to use that in the query to get the User model.

File: app/femr/common/models/PatientEncounterItem.java [hunk 2]
```diff
@@ -37,9 +37,14 @@
...
     private String medicalDateOfVisit;
     private String physicianEmailAddress;
+    private String physicianFullName; //Andrew Change
 
     private String pharmacyDateOfVisit;
     private String pharmacistEmailAddress;
+    private String pharmacistFullName; //Andrew Change
+
+    private String nurseFullName; //Andrew Change
+
```
Comment (kdunlap) [inline app/femr/common/models/PatientEncounterItem.java]: Just a thought: Would it be better to use UserItem here for physician, pharmacist, and nurse?

if we do 

private UserItem physician;
private UserItem pharmacist;
private UserItem nurse;

We'll have access to all the user info, which might be helpful. This definitely isn't necessary and what is here works just fine.

Comment (kdunlap) [inline app/femr/common/models/PatientEncounterItem.java]: We could then replace the below fields

private String physicianEmailAddress;
private String physicianFullName;

private String pharmacistEmailAddress;
private String pharmacistFullName;

private String nurseFullName;
private String nurseEmailAddress;

What do you think?

**Comment by kdunlap** [comment]
We can merge this after the fix to EncounterService.java

Make the changes to your forks branch and this should auto update. I'll keep an eye out and merge whenever thats ready.

**Comment by akrebs1202** [comment]
Ken,

I committed the changes and did another pull request. Do you see the new changes on your end?

**Comment by kdunlap** [comment]
I don't this this updated with your changes.

All you should have to do is push your local changes to your remote branch: improvement-FEMR10-displayUserNames, then the pull request should automatically update. Can you give this a shot again? I'll keep an eye out.

**Comment by akrebs1202** [comment]
Ken,
I pushed the local changes to the remote branch. I received the message stating that "Everything is up-to-date".
-Andrew Krebs


    On Sunday, November 22, 2015 1:04 PM, Ken Dunlap <notifications@github.com> wrote:
 

 I don't this this updated with your changes.All you should have to do is push your local changes to your remote branch: improvement-FEMR10-displayUserNames, then the pull request should automatically update. Can you give this a shot again? I'll keep an eye out.—

**Comment by kdunlap** [comment]
Ok, did you change anything? The branch shows it hasn't been updated in 14 days. I am just looking for the fixes in the EncounterService -- There should only be 1 `nurseQuery`

` ExpressionList<User> nurseQuery1 = QueryProvider.getUserQuery()
                   .where()
                   .eq("email", patientEncounterItem.getNurseFullName()); //Andrew Change
` IUser nurseUser = userRepository.findOne(nurseQuery1);

This code is the issue - it is never going to match anything. the `email` field will never equal `patientEncounterItem.getNurseFullName()`

Is that something you fixed and tried to push up? If we don't resolve this, make sure we do so on Tuesday.

Good luck :-)

**Comment by kevinzurek** [comment]
Andrew -

Based on your GitHub activity, it looks like you might have pushed the changes through your development branch and gotten the up-to-date message. To be reflected in the pull request, you have to make sure you do it on the same branch as the pull request. Scroll up to the top and you'll see the branches:
screenshot from 2015-11-23 19 51 19

-Kevin

**Comment by kdunlap** [comment]
I think this looks good to me. The issue with the nurseQuery was fixed. We should make another task when merging this though to replace the String email/name fields in the PatientEncounterItem class with UserItem

**Pull request merged**